Forta-Bots-Challenge-3/src/utils.ts

```diff
@@ -0,0 +1,96 @@
+import { Finding, getAlerts, sendAlerts } from "forta-agent";
+import { providers, Contract, BigNumber } from "ethers";
+import { createEscrowBalanceFinding, createFinding } from "./finding";
+import { agentId } from "../forta.config.json";
+import { ERC20_ABI } from "./constants";
+
+const chainNames = {
+  1: "Ethereum",
+  10: "Optimism",
+  42161: "Arbitrum",
+};
+
+export type contractAddresses = {
+  l1DaiAddress: string;
+  l2DaiAddress: string;
+  optimismEscrow: string;
+  arbitrumEscrow: string;
+};
+
+export const emitEscrowBalance = async (
+  provider: providers.Provider,
+  blockNumber: number,
+  l1DaiAddress: string,
+  optimismEscrow: string,
+  arbitrumEscrow: string
+) => {
+  const l1DaiContract = new Contract(l1DaiAddress, ERC20_ABI, provider);
+
+  // call L1 DAI contract to get balances of each L2's escrow contract
+  const optimismEscrowBalance = await l1DaiContract.balanceOf(optimismEscrow, { blockTag: blockNumber });
+  const arbitrumEscrowBalance = await l1DaiContract.balanceOf(arbitrumEscrow, { blockTag: blockNumber });
+
+  // create finding
+  const escrowBalanceFinding = createEscrowBalanceFinding(
+    blockNumber.toString(),
+    optimismEscrowBalance,
+    arbitrumEscrowBalance
+  );
+
+  // send alert containing previously declared finding
+  sendAlerts({
+    botId: agentId,
+    finding: escrowBalanceFinding,
+  });
+};
+
+export const checkInvariant = async (
+  provider: providers.Provider,
+  blockNumber: number,
+  findings: Finding[],
+  chainId: number,
+  l1DaiAddress: string
+) => {
+  // create contract instance based on L2 chain bot is currently running on
+  let l2DaiContract: Contract;
+  if (chainId === 10) {
+    l2DaiContract = new Contract(l1DaiAddress, ERC20_ABI, provider);
+  } else if (chainId === 42161) {
+    l2DaiContract = new Contract(l1DaiAddress, ERC20_ABI, provider);
+  } else {
+    throw new Error("You are running the bot in a non supported network");
+  }
```

Comment:
```suggestion
  if (chainId === OP_CHAIN_ID || chainId === ARB_CHAIN_ID) {
    l2DaiContract = new Contract(l2DaiAddress, ERC20_ABI, provider);
  } else {
    throw new Error("You are running the bot in a non supported network");
  }
```
Suggestion to create constants for the Optimism and Arbitrum chain ids like for Ethereum for clarity. Additionally, since their DAI contract addresses are the same, you could combine the `if` checks.

Lastly, it seems the argument `l1DaiAddress` should be `l2DaiAddress`.